.github/workflows/ci.yml

```diff
@@ -10,6 +10,24 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
+  fmt:
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: Swatinem/rust-cache@v2
+      - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
```

Comment:
This will be a rolling version.  It means PR might pass a check when merging but main branch might fail later and need to be updated.

Reply:
Can you explain why will the main branch might fail later if the CI passes?

Reply:
if this runs on 1.68 and passes, later 1.69 comes out with new clippy rules. Then locally update to 1.69, pull main, run clippy and it will fail.  

This can be annoying for contributors as they might have a PR open and rebased on the main branch but now this starts failing when the new tool chain comes out. Either they need to submit a sperate PR and fix then rebase or fix it in same PR to get it in.

If you do release branches this can be a similar issue, where before the fix goes into the branch the formats need to be applied for the new version.

Reply:
Ah I see. What do you recommend to mitigate this issue? I think what we can do is that whenever a new release of clippy comes out (together with a new cargo release), we manually submit a PR to fix the main branch.

Reply:
IMO we should be pinning the version of Rust that we build against. This creates a little bit more work from a maintenance perspective but it allows more control, sets up for repeatable builds and avoids the pitfalls I mentioned above.

Reply:
Done. I've pinned the rust version to 1.69.0